src/main/java/org/folio/dcb/service/impl/BasicPickupLibraryServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+import static org.folio.dcb.domain.dto.TransactionStatus.StatusEnum.OPEN;
+
+@Service("basicPickupLibraryService")
+@RequiredArgsConstructor
+@Log4j2
+public class BasicPickupLibraryServiceImpl implements LibraryService {
+
+  private final TransactionRepository transactionRepository;
+  private final CirculationService circulationService;
+  private final CirculationItemService circulationItemService;
+
+
+  @Override
+  public TransactionStatusResponse createCirculation(String dcbTransactionId, DcbTransaction dcbTransaction, String pickupServicePointId) {
+    return null;
+  }
+
+  @Override
+  public void updateStatusByTransactionEntity(TransactionEntity transactionEntity) {
+    log.debug("updateTransactionStatus:: Received checkIn event for itemId: {}", transactionEntity.getItemId());
+    CirculationItemRequest circulationItemRequest = circulationItemService.fetchItemById(transactionEntity.getItemId());
+    if (OPEN == transactionEntity.getStatus() && AWAITING_PICKUP == circulationItemRequest.getStatus().getName()) {
+      updateTransactionEntity(transactionEntity, TransactionStatus.StatusEnum.AWAITING_PICKUP);
+    } else if (TransactionStatus.StatusEnum.AWAITING_PICKUP == transactionEntity.getStatus() && CHECKED_OUT == circulationItemRequest.getStatus().getName()) {
+      updateTransactionEntity(transactionEntity, ITEM_CHECKED_OUT);
+    } else if(ITEM_CHECKED_OUT == transactionEntity.getStatus()){
+      log.info("updateStatusByTransactionEntity:: Updated item status from {} to {}",
+        transactionEntity.getStatus().getValue(), TransactionStatus.StatusEnum.ITEM_CHECKED_IN.getValue());
+      updateTransactionEntity(transactionEntity, TransactionStatus.StatusEnum.ITEM_CHECKED_IN);
+    } else {
+      log.info("updateStatusByTransactionEntity:: Item status is {}. So status of transaction is not updated",
+        circulationItemRequest.getStatus().getName());
+    }
+
+  }
+
+  @Override
+  public void updateTransactionStatus(TransactionEntity dcbTransaction, TransactionStatus transactionStatus) {
+    log.debug("updateTransactionStatus:: Updating dcbTransaction {} to status {} ", dcbTransaction, transactionStatus);
+    var currentStatus = dcbTransaction.getStatus();
+    var requestedStatus = transactionStatus.getStatus();
+    if (CREATED == currentStatus && OPEN == requestedStatus) {
+      log.info("updateTransactionStatus:: Checking in item by barcode: {} ", dcbTransaction.getItemBarcode());
+      //Random UUID for servicePointId.
+      circulationService.checkInByBarcode(dcbTransaction, UUID.randomUUID().toString());
+      updateTransactionEntity(dcbTransaction, requestedStatus);
+    } else if (ITEM_CHECKED_IN == currentStatus && CLOSED == requestedStatus) {
+      updateTransactionEntity(dcbTransaction, requestedStatus);
+    } else {
+      String errorMessage = String.format("updateTransactionStatus:: status update from %s to %s is not implemented", currentStatus, requestedStatus);
+      log.warn(errorMessage);
```

Comment:
I think errorMessage string object can be avoided and the string format can be printed directly in warn logs.